opm/parser/eclipse/EclipseState/Schedule/Completion.cpp

```diff
@@ -136,12 +138,22 @@ namespace Opm {
        disentangled, and each completion is returned separately.
     */
 
-    std::pair<std::string , std::vector<CompletionPtr> > Completion::completionsFromCOMPDATRecord( const DeckRecord& compdatRecord ) {
+    inline std::vector< CompletionPtr >
+    fromCOMPDAT( const DeckRecord& compdatRecord, const Well& well ) {
+
         std::vector<CompletionPtr> completions;
-        std::string well = compdatRecord.getItem("WELL").getTrimmedString(0);
+
         // We change from eclipse's 1 - n, to a 0 - n-1 solution
-        int I = compdatRecord.getItem("I").get< int >(0) - 1;
-        int J = compdatRecord.getItem("J").get< int >(0) - 1;
+        // I and J can be defaulted with 0 or *, in which case they are fetched
+        // from the well head
+        const auto& itemI = compdatRecord.getItem( "I" );
+        const auto defaulted_I = itemI.defaultApplied( 0 ) || itemI.get< int >( 0 ) == 0;
```

Comment:
Nah, it's not something to just "suck up" to be honest, we should rather figure out something that works for us based on its merits. I (obviously) think the ternary has merit **because** it's an expression, and I have yet to be convinced that the alternative provides any benefits, but I'm open to be persuaded.